### ISSUE TITLE
Training does not decrease error given a custom loss function

### ISSUE THREAD

**Issue by ajsanjoaquin** [PyTorchLightning/pytorch-lightning#9324]
### Discussed in https://github.com/PyTorchLightning/pytorch-lightning/discussions/9323

<div type='discussions-op-text'>

<sup>Originally posted by **ajsanjoaquin** September  4, 2021</sup>
Hello,

I implemented this loss function from Section 2.1 of [Right for the Right Reasons by Ross, et al., 2017](https://arxiv.org/pdf/1703.03717.pdf)

![image](https://user-images.githubusercontent.com/42886152/132091707-436d7176-7404-435d-86da-3a3abf7cbb02.png)

Training updates normally when using vanilla torch, but it does not decrease the loss when I implemented the same setup in Pytorch Lightning. Specifically, I use a zero matrix (A term in the equation) as an input, which makes the custom loss equivalent to cross entropy.

[Here is a notebook to reproduce the error](https://colab.research.google.com/drive/1lFbFmtQcXlO_QVOOBC7hZWrxuJeJSVee?usp=sharing), and show that the same setup works in vanilla pytorch. Hoping that I just missed a step somewhere and that someone can point it out. 

Thank you!
</div>
